src/java/org/apache/cassandra/cql3/ConstraintScalarCondition.java

```diff
@@ -0,0 +1,143 @@
+        {
+            this.param = param;
+            this.relationType = relationType;
+            this.term = term;
+        }
+
+        public ConstraintScalarCondition prepare()
+        {
+            return new ConstraintScalarCondition(param, relationType, term);
+        }
+    }
+
+    public ConstraintScalarCondition(ColumnIdentifier param, Operator relationType, String term)
+    {
+        this.param = param;
+        this.relationType = relationType;
+        this.term = term;
+    }
+
+    public void checkCondition(Map<String, String> columnValues, ColumnMetadata columnMetadata, TableMetadata tableMetadata)
+    {
+        int columnValue = Integer.parseInt(columnValues.get(param.toString()));
+        int sizeConstraint = Integer.parseInt(term);
+
+        switch (relationType)
+        {
+            case EQ:
+                if (columnValue != sizeConstraint)
+                    throw new ConstraintViolationException(param + " value length should be exactly " + sizeConstraint);
+                break;
+            case NEQ:
+                if (columnValue == sizeConstraint)
+                    throw new ConstraintViolationException(param + " value length different than " + sizeConstraint);
+                break;
+            case GT:
+                if (columnValue <= sizeConstraint)
+                    throw new ConstraintViolationException(param + " value length should be larger than " + sizeConstraint);
+                break;
+            case LT:
+                if (columnValue >= sizeConstraint)
+                    throw new ConstraintViolationException(param + " value length should be smaller than " + sizeConstraint);
+                break;
+            case GTE:
+                if (columnValue < sizeConstraint)
+                    throw new ConstraintViolationException(param + " value length should be larger or equal than " + sizeConstraint);
+                break;
+            case LTE:
+                if (columnValue > sizeConstraint)
+                    throw new ConstraintViolationException(param + " value length should be smaller or equala than " + sizeConstraint);
+                break;
+            default:
+                throw new ConstraintViolationException("Invalid relation type: " + relationType);
+        }
+    }
+
+    public void validateCondition(ColumnMetadata columnMetadata, TableMetadata tableMetadata)
+    {
+        if (!(tableMetadata.getColumn(param).type instanceof org.apache.cassandra.db.marshal.NumberType))
+            throw new ConstraintViolationException(param + " is not a number");
+    }
+
+    @Override
+    public String toString()
+    {
+        return String.format("%s %s %s", param, relationType, term);
+    }
+
+    public void serialize(ConstraintCondition constraintFunctionCondition, DataOutputPlus out, Version version) throws IOException
+    {
+        serializer.serialize((ConstraintScalarCondition) constraintFunctionCondition, out, version);
```

Comment:
just saying it is interesting to see that we need to cast here. I would expect that the serializer is parameterized already on creation so is all type safe and stuff

Reply:
I do not think the first parameter is ever needed.